### PR TITLE
Aux/marketplace

### PR DESCRIPTION
**Pull request by sshaman1101**
This PR adds a simple centralized marketplace implementation.

### CODE REVIEW

File: insonmnia/marketplace/server.go [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+		return errSlotIsNil
+	}
+
+	if slot.Resources == nil {
+		return errResourcesIsNil
+	}
+	return nil
+}
+
+func (in *inMemOrderStorage) compareTime(slot *pb.Slot, order *pb.Order) bool {
```
Comment (3Hren) [inline insonmnia/marketplace/server.go]: Order contains a slot, which means that you're comparing a slot with another slot. Then maybe better to represent it explicitly?

`func (s *Slot) compareTime(other *Slot) bool` and so on

Comment (3Hren) [reply insonmnia/marketplace/server.go]: Of course you need a wrapper type: `type Slot pb.Slot`.

**Comment by 3Hren** [comment]
I need the same code for comparing slots, so I'm going to just wait until you complete :)

File: insonmnia/marketplace/server.go [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+	errOrderIsNil        = errors.New("Order cannot be nil")
+	errSlotIsNil         = errors.New("Order slot cannot be nil")
+	errResourcesIsNil    = errors.New("Slot resources cannot be nil")
+	errStartTimeAfterEnd = errors.New("Start time is after end time")
+	errStartTimeRequired = errors.New("Start time is required")
+	errEndTimeRequired   = errors.New("End time is required")
+)
+
+type OrderStorage interface {
+	GetOrders(slot *pb.Slot) ([]*pb.Order, error)
```
Comment (3Hren) [inline insonmnia/marketplace/server.go]: Great idea: make `Order` wrapper, which is constructed from `pb.Order` and performs all required checks (like required fields). Otherwise all other implementation will suffer from doing the same things.

Protobuf sucks, yeah.

Make in a separate PR.

**Comment by 3Hren** [comment]
Also conflicts. Rebase and merge.

**Comment by 3Hren** [comment]
Emm, why did you add locator code here? It is out of the scope and is discussed in a separate PR.

**Comment by sshaman1101** [comment]
Oh, merged with my local changes and accidentally pushed.
Commits from the locator branch were removed!
